controller_manager:
  ros__parameters:
    update_rate: 500  # 500 Hz control loop

    stepper_trajectory_controller:
      type: joint_trajectory_controller/JointTrajectoryController

    joint_state_broadcaster:
      type: joint_state_broadcaster/JointStateBroadcaster


stepper_trajectory_controller:
  ros__parameters:
    joints:
      - stepper_joint
    
    command_interfaces:
      - velocity
    
    state_interfaces:
      - position
      - velocity
    
    open_loop_control: true
    
    interpolation_method: none
    
    allow_partial_joints_goal: false
    
    constraints:
      stopped_velocity_tolerance: 0.01
      goal_time: 0.0
      stepper_joint:
        trajectory: 100.0
        goal: 100.0
    
    gains:
      stepper_joint:
        p: 0.0
        i: 0.0
        d: 0.0

joint_state_broadcaster:
  ros__parameters:
    joints:
      - stepper_joint
